### DISCUSSION TITLE
How to get a list of all Pages by language

### DISCUSSION THREAD

**Discussion by carlosnasillo** [wagtail/wagtail-localize#289] [Q&A] [answered]
In the `get_context()` for my `BlogPostsListView(Page)` I have the following method to get a list of all the available posts.

The **issue** I'm having is that this returns duplicates (English, Spanish) instances for all the posts. Is there a way to filter by a specific language? or similar behaviour so it respects the current locale set by the user (ie. `/es/` --> Spanish instances only) 

```
class BlogPostsListView(Page):
    
   def get_context(self, request, *args, **kwargs):
        context = super().get_context(request, *args, **kwargs)
        context["featured"] = BlogPost.objects.live().public().last()
        all_posts = BlogPost.objects.live().public().order_by("-date")
        paginator = Paginator(all_posts, 6)

        # Try to get the ?page=x value
        page = request.GET.get("page")
        try:
            # If the page exists and the ?page=x is an int
            posts = paginator.page(page)
        except PageNotAnInteger:
            # If the ?page=x is not an int; show the first page
            posts = paginator.page(1)
        except EmptyPage:
            # If the ?page=x is out of range (too high most likely)
            # Then return the last page
            posts = paginator.page(paginator.num_pages)

        context["posts"] = posts
        return context
```

Thanks. This project looks mega promising.

**Comment by kaedroho** [comment, marked as answer]
You can filter pages by their ``locale`` field. To get the current actively locale, use [``Locale.get_active()``](https://docs.wagtail.io/en/stable/reference/pages/model_reference.html?highlight=Locale.get_active#wagtail.core.models.Locale.get_active)

For example:

```python
current_locale_posts = BlogPost.objects.filter(locale=Locale.get_active())
```

**Reply by carlosnasillo** [reply]
Thanks Karl, this is super helpful!
